client.go

```diff
@@ -47,6 +47,9 @@ var (
 	defaultRetryWaitMax = 30 * time.Second
 	defaultRetryMax     = 4
 
+	// defaultLogger is the logger provided with defaultClient
+	defaultLogger = log.New(os.Stderr, "", log.LstdFlags)
```

Comment:
I only added this to keep the code DRY and re-usable since I used the same logger in the tests (see [here](https://github.com/hashicorp/go-retryablehttp/pull/67/files#diff-51565ad50d7426213c2f33579f6f0e71R299)). Happy to copy and paste if you prefer that